lib/site-builder.js

```diff
@@ -154,11 +154,26 @@ SiteBuilder.launchBuilder = function(info, branch, builderConfig) {
         return err ? Promise.reject(err) : Promise.resolve();
       });
   };
-  return builder.build()
+  return createBuildDestinationDir(builder.configHandler.buildDestination)
+    .then(function() {
+      return builder.build();
+    })
     .then(finishBuild, finishBuild)
     .then(migrateLog, migrateLog);
 };
 
+function createBuildDestinationDir(buildDestinationDir) {
+  return new Promise(function(resolve, reject) {
+    fs.mkdir(buildDestinationDir, function(err) {
+      if (err && err.code !== 'EEXIST') {
+        return reject(new Error('Failed to create ' + buildDestinationDir +
+          ': ' + err.message));
+      }
+      resolve();
+    });
```

Comment:
Not sure it simplified it that much per se, but it seems a robust and widely-used package, so done. (Also tweaked the error message slightly.)